packages/dialog/src/dialog.tsx

```diff
@@ -14,23 +14,33 @@ import type {
   DialogContentProps,
   DialogOverlayProps,
   DialogPortalProps,
-  DialogRootProps,
+  DialogRootProps as IDialogRootProps,
 } from './types';
+import { useControllableState } from '@rnr/hooks';
 
-interface RootContext extends DialogRootProps {
+interface DialogRootProps extends Partial<IDialogRootProps> {
+  defaultOpen?: boolean;
+}
+
+interface RootContext extends IDialogRootProps {
   nativeID: string;
 }
```

Comment:
Instead of extending the `DialogRootProps` with a partial, can you add the `defaultOpen` props to the `DialogRootProps` in the `./types`?

Reference: A

Reply:
i thought letting default DialogRootProps( IDialogRootProps ) required for the Context would be a good idea

Reply:
Good point, in that case I would do as Reference A and then I would write context type as

```tsx
interface RootContext extends Required<Omit<DialogRootProps, 'defaultOpen'>> {
    nativeID: string
}
```
or
```tsx
interface RootContext {
    open: boolean;
    onOpenChange: (value: boolean) => void;
    nativeID: string
}
```

The idea is that the component props live in the `types.ts` file as is it important that the default and the web primitives share the same prop types.

Reply:
Now that I think about it, the context type should probably live in the `types.ts` file as well since they `useRootContext` is exported from both. Applying that to all primitives is out of scope so for these components, you can place the context type where you want.

Reply:
OK on it